File: lab/python/binarios/basen.py
def convert_to_base(number, base):
    if number < 2 and number > 16:
        return ""

    digits = "0123456789ABCDEF"
    result = ""
    while number > 0:
        resto = number % base
        result = digits[resto] + result
        number //= base

    return result


print(convert_to_base(25, 2))    # 11001
print(convert_to_base(25, 3))    # 221
print(convert_to_base(255, 16))  # FF
print(convert_to_base(100, 8))   # 144

File: lab/javascript/binarios/basen.js
function toBaseN(number, base) {
  const digits = "0123456789ABCDEF";
  if (number === 0) return "0";
  let result = "";

  while (number > 0) {
    let resto = number % base;
    result = digits[resto] + result;
    number = Math.floor(number / base);
  }

  return result;
}

console.log(toBaseN(25, 2));   // 11001
console.log(toBaseN(25, 3));   // 221
console.log(toBaseN(255, 16)); // FF
console.log(toBaseN(100, 8));  // 144
